thriftpy/thrift.py

```diff
@@ -192,9 +192,28 @@ def _negotiation(self):
             result.read(self._iprot)
             self._iprot.read_message_end()
 
-    def _send(self, _api, **kwargs):
+    def _req(self, *args, **kwargs):
+        try:
+            exception = None
+            res = super(TTrackedClient, self)._req(*args, **kwargs)
+        except Exception as e:
+            exception = e
+
         if self._upgraded:
             header = trace.thrift.RequestHeader()
+            header.read(self._iprot)
+            self._iprot.read_message_end()
+            header.status = not exception
+            self.track_handler.record_header(header)
```

Comment:
Client read the header and do record stuffs. Take a look at the **status** here.